File: examples/rust/transport_mock.rs
//! Configure a Wyrd queue with the in-memory mock transport.
//!
//! Run with:
//!     cargo run -p wyrd-rust-examples --bin transport_mock

use wyrd_client::transport::{MockConfig, QueueConfig, TransportConfig};

fn main() -> anyhow::Result<()> {
    let mock = MockConfig {
        label: "demo".to_string(),
        fail_on_flush: None,
    };
    let queue = QueueConfig {
        transport: TransportConfig::Mock(mock),
        flush_max_rows: 100,
        flush_interval_ms: 100,
        channel_capacity: 10,
        sample_ratio: None,
    };
    queue.validate()?;

    let json = serde_json::to_string_pretty(&queue)?;
    let round_trip: QueueConfig = serde_json::from_str(&json)?;
    anyhow::ensure!(round_trip == queue, "queue config did not round-trip");
    println!("{json}");
    Ok(())
}

File: sdks/wyrd-sdk-python/examples/transport_mock.py
"""Configure the in-memory mock transport as JSON-compatible config.

Run with:
    python sdks/wyrd-sdk-python/examples/transport_mock.py
"""

from __future__ import annotations

import json


def main() -> None:
    mock = {
        "label": "demo",
        "fail_on_flush": None,
    }
    queue = {
        "transport": {"transport": "mock", "params": mock},
        "flush_max_rows": 100,
        "flush_interval_ms": 100,
        "channel_capacity": 10,
    }
    encoded = json.dumps(queue, indent=2, sort_keys=True)
    assert json.loads(encoded) == queue
    print(encoded)


if __name__ == "__main__":
    main()
